src/one-source/buffer-with-time-or-count.js

```diff
@@ -0,0 +1,63 @@
+const {createStream, createProperty} = require('../patterns/one-source');
+
+const mixin = {
+
+  _init({wait, count, flushOnEnd}) {
+    this._wait = wait;
+    this._count = count;
+    this._flushOnEnd = flushOnEnd;
+    this._intervalId = null;
+    this._$onTick = () => this._flush();
+    this._buff = [];
+  },
+
+  _free() {
+    this._$onTick = null;
+    this._buff = null;
+  },
+
+  _flush() {
+    if (this._buff !== null && this._buff.length !== 0) {
```

Comment:
Let's not check for `length` here, and simply emit empty arrays when buffer is empty. This is necessary in some cases to know when buffer flushes, and it's easy to filter out those empty flushes with `.filter(x => x.length > 0)` (if the code that handles arrays sensitive to their length).

We already do this in all other buffering methods.